apple/testing/default_runner/ios_xctestrun_runner.template.sh

```diff
@@ -0,0 +1,254 @@
+  xctestrun_env+="<key>XCInjectBundleInto</key><string>$(escape "__TESTHOST__/$test_host_name.app/$test_host_name")</string>"
+else
+  xctestrun_test_host_path="__PLATFORMS__/iPhoneSimulator.platform/Developer/Library/Xcode/Agents/xctest"
+  xctestrun_test_host_based=false
+fi
+
+profraw=$(mktemp)
+
+sanitizer_dyld_env=""
+readonly sanitizer_root="$test_tmp_dir/$test_bundle_name.xctest/Frameworks"
+for sanitizer in "$sanitizer_root"/libclang_rt.*.dylib; do
+  if [[ -n "$sanitizer_dyld_env" ]]; then
+    sanitizer_dyld_env="$sanitizer_dyld_env:"
+  fi
+  sanitizer_dyld_env="${sanitizer_dyld_env}${sanitizer}"
+done
+
+xctestrun_libraries="__PLATFORMS__/iPhoneSimulator.platform/Developer/usr/lib/libXCTestBundleInject.dylib"
+if [[ -n "$sanitizer_dyld_env" ]]; then
+  xctestrun_libraries="${xctestrun_libraries}:${sanitizer_dyld_env}"
+fi
+
+readonly xctestrun_file="$test_tmp_dir/tests.xctestrun"
+/usr/bin/sed \
+  -e "s@BAZEL_INSERT_LIBRARIES@$xctestrun_libraries@g" \
+  -e "s@BAZEL_TEST_BUNDLE_PATH@__TESTROOT__/$test_bundle_name.xctest@g" \
+  -e "s@BAZEL_TEST_ENVIRONMENT@$xctestrun_env@g" \
+  -e "s@BAZEL_TEST_HOST_BASED@$xctestrun_test_host_based@g" \
+  -e "s@BAZEL_TEST_HOST_PATH@$xctestrun_test_host_path@g" \
+  -e "s@BAZEL_TEST_ORDER_STRING@%(test_order)s@g" \
+  -e "s@BAZEL_COVERAGE_PROFRAW@$profraw@g" \
+  -e "s@BAZEL_COVERAGE_OUTPUT_DIR@$test_tmp_dir@g" \
+  "%(xctestrun_template)s" > "$xctestrun_file"
+
+id="$("./%(simulator_creator.py)s" "%(os_version)s" "%(device_type)s")"
+test_exit_code=0
+testlog=$(mktemp)
+
+
+test_file=$(file "$test_tmp_dir/$test_bundle_name.xctest/$test_bundle_name")
+intel_simulator_hack=false
+if [[ $(arch) == arm64 && "$test_file" != *arm64* ]]; then
+  intel_simulator_hack=true
+fi
+
+if [[ -n "$test_host_path" || -n "${CREATE_XCRESULT_BUNDLE:-}" ]]; then
+  if [[ -z "$test_host_path" && "$intel_simulator_hack" == true ]]; then
+    echo "error: running x86_64 tests on arm64 macs with CREATE_XCRESULT_BUNDLE requires a test host" >&2
+    exit 1
+  fi
+
+  args=(
+    -destination "id=$id" \
+    -destination-timeout 15 \
+    -xctestrun "$xctestrun_file" \
+  )
+
+  readonly result_bundle_path="$TEST_UNDECLARED_OUTPUTS_DIR/tests.xcresult"
+  # TEST_UNDECLARED_OUTPUTS_DIR isn't cleaned up with multiple retries of flaky tests
+  rm -rf "$result_bundle_path"
+  if [[ -n "${CREATE_XCRESULT_BUNDLE:-}" ]]; then
+    args+=(-resultBundlePath "$result_bundle_path")
+  fi
+
+  xcodebuild test-without-building "${args[@]}" \
+    2>&1 | tee -i "$testlog" | (grep -v "One of the two will be used" || true) \
+    || test_exit_code=$?
+else
+  platform_developer_dir="$(xcode-select -p)/Platforms/iPhoneSimulator.platform/Developer"
+  xctest_binary="$platform_developer_dir/Library/Xcode/Agents/xctest"
+  test_file=$(file "$test_tmp_dir/$test_bundle_name.xctest/$test_bundle_name")
+  if [[ "$intel_simulator_hack" == true ]]; then
+    sliced_xctest_binary=/tmp/xctest_intel
```

Comment:
Can this be in the output_base instead, similar to global module caches? Seems odd to write outside of it.